OracleSOASuite/dockerfiles/12.2.1.3/README.md

```diff
@@ -0,0 +1,170 @@
+SOA on Docker
+=============
+
+Sample Docker configurations to facilitate installation, configuration, and environment setup for Docker users. This project includes quick start dockerfiles for Oracle SOA 12.2.1.3.0 based on Oracle Linux 7, Oracle JRE 8 (Server) and Oracle Fusion Middleware Infrastructure 12.2.1.3.0.
+
+At the end of this configuration there may be 3 containers running: 
+1. Oracle Database Container (Only when container based Database is used)
+2. Oracle Weblogic Admin Server Container 
+3. Oracle Weblogic Managed Server Container (SOA Server)
+
+The containers will be connected using a Docker User Defined network. 
+
+## Create a User Defined network
+
+In this configuration creation of a user defined network will enable the communication between the containers just using container names. For this setup we will use a user defined network using bridge driver.
+
+Create a user defined network using the bridge driver by executing the following command :
+
+    $ docker network create -d bridge <some name>
+   
+Sample command:
+
+    $ docker network create -d bridge SOANet
+
+# Mount a host directory as a data volume
+
+Data volumes are designed to persist data, independent of the container’s lifecycle. The default location of the volume in container is under `/var/lib/docker/volumes`. There is an option to mount a directory from the host into a container as volume. In this project we will use that option for the data volume. The volume will be used to store Database datafiles and WebLogic Server domain files. This volume will be created on the host at `/scratch/DockerVolume/SOAVolume/`. Since the volume is created as "root" user, provide read/write/execute permissions to "oracle" user (by providing permissions to "others"), as all operations inside the container happens with "oracle" user login.
+
+To determine if a user already exists on your host system with uid:gid of 1000, run:
+
+    # getent passwd 1000
+
+If that returns a username (which is the first field), use that user for the `useradd` command below. If not, create the `oracle` user manually. 
```

Comment:
If `getent` returns a username, then a user with a `uid` of `1000` already exists and any attempt to run `useradd` with the same name should fail or cause inconsistent results (depending on the host OS).